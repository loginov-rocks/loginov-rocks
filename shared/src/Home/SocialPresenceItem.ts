export interface SocialPresenceItem {
  title: string;
  url: string;
  url2?: string;
}
