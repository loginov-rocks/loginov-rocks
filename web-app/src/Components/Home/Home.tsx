import { GitHubData } from '@loginov-rocks/loginov-rocks-shared';
import * as React from 'react';

import { SocialPresence } from 'Components/SocialPresence';
import { WorksList } from 'Components/WorksList';
import { WEB_APP_S3_GITHUB_FILE_KEY } from 'Constants';
import { HomeData } from 'Data/Interfaces/HomeData';
import { GitHubData as GetGitHubData } from 'Lib/GitHubData';

const getGitHubData = new GetGitHubData({ url: `/${WEB_APP_S3_GITHUB_FILE_KEY}` });

interface Props {
  data: HomeData;
}

export const Home: React.FunctionComponent<Props> = ({ data }) => {
  const [gitHubData, setGitHubData] = React.useState<GitHubData | null>(null);

  React.useEffect(() => {
    getGitHubData.get()
      .then((_gitHubData) => {
        setGitHubData(_gitHubData);
      });
  }, []);

  return (
    <>
      <h1>Hello!</h1>
      <p>
        I am
        {' '}
        <strong>Danila</strong>
        , nice to e-meet you!
      </p>
      <>
        <h2>Social Presence</h2>
        <SocialPresence items={data.socialPresenceItems} />
      </>
      <>
        <h2>Open Source</h2>
        <>
          <h3>Projects</h3>
          <WorksList gitHubRepos={gitHubData ? gitHubData.repos : []} works={data.openSource.projects} />
        </>
        <>
          <h3>Libraries</h3>
          <WorksList gitHubRepos={gitHubData ? gitHubData.repos : []} works={data.openSource.libraries} />
        </>
      </>
    </>
  );
};
