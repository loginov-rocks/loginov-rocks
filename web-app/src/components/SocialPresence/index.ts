import { SocialPresence } from './SocialPresence';

export { SocialPresence };
