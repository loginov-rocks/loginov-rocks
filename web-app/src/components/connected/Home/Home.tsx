import * as React from 'react';

import { CmsConnectedProps } from 'cms/interfaces/CmsConnectedProps';
import { CmsEntry } from 'cms/interfaces/CmsEntry';

interface Props extends CmsConnectedProps {
  sections: CmsEntry[];
  title: string;
}

export const Home: React.FC<Props> = ({ render, sections, title }) => (
  <>
    <h1>{title}</h1>
    {render(sections)}
  </>
);
