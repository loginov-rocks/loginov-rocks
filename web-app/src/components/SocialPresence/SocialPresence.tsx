import { SocialPresenceItem } from '@loginov-rocks/loginov-rocks-shared';
import * as React from 'react';

import { SOCIAL_PRESENCE_DRIVE2_TITLE } from 'Constants';

import { Link } from 'components/Link';

interface Props {
  items: SocialPresenceItem[];
}

export const SocialPresence: React.FunctionComponent<Props> = ({ items }) => (
  <ul>
    {items.map((item, index) => {
      let content;

      if (item.title === SOCIAL_PRESENCE_DRIVE2_TITLE) {
        content = (
          <>
            <Link href={item.url}>{item.title}</Link>
            {' '}
            &mdash; my Toyota Chaser GX71 blog, originally
            {' '}
            <Link href={item.url2 as string}>in Russian</Link>
          </>
        );
      } else {
        content = (
          <Link href={item.url}>{item.title}</Link>
        );
      }

      return (
        // eslint-disable-next-line react/no-array-index-key
        <li key={index}>
          {content}
        </li>
      );
    })}
  </ul>
);
