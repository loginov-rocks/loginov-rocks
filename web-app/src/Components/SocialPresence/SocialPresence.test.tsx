/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import * as homeData from 'Data/Home.json';

import { SocialPresence } from './SocialPresence';

it('matches snapshot', () => {
  const wrapper = shallow(<SocialPresence items={homeData.socialPresenceItems} />);

  expect(wrapper).toMatchSnapshot();
});
