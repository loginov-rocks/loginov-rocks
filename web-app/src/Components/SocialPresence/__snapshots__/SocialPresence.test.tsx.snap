// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ul>
  <li
    key="0"
  >
    <Link
      href="https://www.linkedin.com/in/loginov-rocks/"
    >
      LinkedIn
    </Link>
  </li>
  <li
    key="1"
  >
    <Link
      href="https://github.com/loginov-rocks"
    >
      GitHub
    </Link>
  </li>
  <li
    key="2"
  >
    <Link
      href="https://loginov-rocks.medium.com/"
    >
      Medium
    </Link>
  </li>
  <li
    key="3"
  >
    <Link
      href="https://www.drive2.com/r/toyota/chaser/288230376151952785/"
    >
      DRIVE2
    </Link>
     
    â€” my Toyota Chaser GX71 blog, originally
     
    <Link
      href="https://www.drive2.ru/r/toyota/chaser/288230376151952785/"
    >
      in Russian
    </Link>
  </li>
  <li
    key="4"
  >
    <Link
      href="https://www.instagram.com/loginov_rocks/"
    >
      Instagram
    </Link>
  </li>
  <li
    key="5"
  >
    <Link
      href="https://www.facebook.com/danila.rocks"
    >
      Facebook
    </Link>
  </li>
  <li
    key="6"
  >
    <Link
      href="https://twitter.com/loginov_rocks"
    >
      Twitter
    </Link>
  </li>
  <li
    key="7"
  >
    <Link
      href="https://www.npmjs.com/~loginov-rocks"
    >
      npm
    </Link>
  </li>
  <li
    key="8"
  >
    <Link
      href="https://stackoverflow.com/users/5021580/danila-loginov"
    >
      Stack Overflow
    </Link>
  </li>
  <li
    key="9"
  >
    <Link
      href="https://vk.com/danilaloginov"
    >
      vk
    </Link>
  </li>
  <li
    key="10"
  >
    <Link
      href="https://www.youtube.com/c/DanilaLoginov"
    >
      YouTube
    </Link>
  </li>
</ul>
`;
